ssl-config/src/main/java/com/palantir/remoting/ssl/SslConfiguration.java

```diff
@@ -28,18 +30,51 @@
 @Value.Style(visibility = ImplementationVisibility.PACKAGE)
 public abstract class SslConfiguration {
 
-    @Value.Parameter
-    public abstract TrustStoreConfiguration trust();
+    private static final String DEFAULT_STORE_TYPE = "JKS";
 
-    @Value.Parameter
-    public abstract Optional<KeyStoreConfiguration> key();
+    public abstract Path trustStorePath();
 
-    public static SslConfiguration of(TrustStoreConfiguration trust) {
-        return ImmutableSslConfiguration.of(trust, Optional.<KeyStoreConfiguration>absent());
+    @SuppressWarnings("checkstyle:designforextension")
+    @Value.Default
+    public String trustStoreType() {
+        return DEFAULT_STORE_TYPE;
     }
 
-    public static SslConfiguration of(TrustStoreConfiguration trust, KeyStoreConfiguration key) {
-        return ImmutableSslConfiguration.of(trust, Optional.of(key));
+    public abstract Optional<Path> keyStorePath();
+
+    public abstract Optional<String> keyStorePassword();
+
+    @SuppressWarnings("checkstyle:designforextension")
+    @Value.Default
+    public String keyStoreType() {
+        return DEFAULT_STORE_TYPE;
+    }
+
+    // alias of the key that should be used in the key store.
+    // If absent, first entry returned by key store is used.
+    public abstract Optional<String> keyStoreKeyAlias();
+
+    @Value.Check
+    protected final void check() {
+        Preconditions.checkArgument(
+                keyStorePath().isPresent() == keyStorePassword().isPresent(),
+                "keyStorePath and keyStorePassword must both be present or both be absent");
+
+        Preconditions.checkArgument(
+                !keyStoreKeyAlias().isPresent() || keyStorePath().isPresent(),
+                "keyStoreKeyAlias must be present if keyStorePath is present");
```

Comment:
Oops, sorry, message is incorrect here. What you're saying is correct -- this is meant to throw an error if `keyStoreKeyAlias` is specified when `keyStorePath` is not present.